src/models/product.js

```diff
@@ -0,0 +1,75 @@
+const mongoose = require('mongoose');
+
+const productSchema = new mongoose.Schema(
+  {
+    title: {
+      type: String,
+      maxLength: 150,
+      required: [true, 'Title cannot be empty!'],
+    },
+    description: {
+      type: String,
+      required: [true, 'Description cannot be empty!'],
+    },
+    price: {
+      type: Number,
+      required: [true, 'Price cannot be empty!'],
+    },
+    category: {
+      type: String,
+      required: [true, 'Please type in a category!'],
+    },
+    coverImage: {
+      type: String,
+      required: [true, 'Cover image cannot be empty!'],
+    },
+    images: {
+      type: [String],
+    },
+    type: {
+      type: [String],
+      enum: {
+        values: ['Product', 'Service'],
+        message: '{VALUE} is not supported!',
+      },
+      required: [true, 'Please select the type of the product!'],
+    },
+    location: {
+      type: String,
+      required: [true, 'Location cannot be empty!'],
+      maxLength: 50,
+    },
+    status: {
+      type: [String],
```

Comment:
the `status` field is an array here, should be a string.
Note: surrounding a field type with `[]` makes it an array.